engine/src/flutter/lib/ui/painting/rsuperellipse.cc

```diff
@@ -6,54 +6,67 @@
 
 #include "flutter/fml/logging.h"
 #include "third_party/tonic/logging/dart_error.h"
-#include "third_party/tonic/typed_data/typed_list.h"
 
-using flutter::RSuperellipse;
+namespace flutter {
 
-namespace tonic {
+using impeller::Scalar;
 
-// Construct an DlRoundSuperellipse from a Dart RSuperellipse object.
-// The Dart RSuperellipse is a Float32List containing
-//   [left, top, right, bottom, xRadius, yRadius]
-RSuperellipse DartConverter<flutter::RSuperellipse>::FromDart(
-    Dart_Handle value) {
-  Float32List buffer(value);
+IMPLEMENT_WRAPPERTYPEINFO(ui, RSuperellipse);
 
-  RSuperellipse result;
-  result.is_null = true;
-  if (buffer.data() == nullptr) {
-    return result;
+RSuperellipse::RSuperellipse(const tonic::Float64List& values) {
```

Comment:
I'm not sure what the usage patterns will look like here, but passing in a dart typed data (and then acquiring a lock in the dart GC) will be a lot slower than passing in ~12 floating points.

Also I think should should truncate the 32 bit during construction.

Reply:
Ok, I will make it a 12-argument function.

> Also I think should should truncate the 32 bit during construction.

The reason why I kept 64 bit here is because the Dart app might query these values (such as `get double top`). Since the Dart app assigned 64 bit to the class, it should probably expect the same 64 bit is stored and returned (even though all later calculation only uses 32 bits.) 

There are three options I can think of:
1. (Current) Store and return 64 bit, even though later calculation only uses 32 bits.
2. Store and return 32 bit. The Dart app probably won't care anyway.
3. Make the Dart class store a Dart double (which will reduce a lot of LOC!), and the C++ class store Scalar separately.

Which would you recommend?

Reply:
After discussion we've agreed to go with option 3, which has the best benefit with neglegible memory overhead.